File: solutions/csharp/hamming/1/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length)
            throw new ArgumentException("Arguments must both have the same length.");

        var original  = firstStrand .ToCharArray();
        var newStrand = secondStrand.ToCharArray();
        var count     = 0;

        for (var i = 0; i < original.Length; i++)
            if (original[i] != newStrand[i])
                count++;

        return count;
    }
}

File: solutions/go/hamming/1/hamming.go
package hamming

import "fmt"

func Distance(a, b string) (int, error) {
    if len(a) != len(b) {
        return -1, fmt.Errorf("Arguments must be the same length.")
    }

    count := 0

    for i, _ := range a {
        if (a[i] != b[i]) {
            count++
        }
    }

    return count, nil
}
